nemo/collections/nlp/models/intent_slot_classification_model.py

```diff
@@ -0,0 +1,264 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Optional
+
+import torch
+from omegaconf import DictConfig
+from pytorch_lightning import Trainer
+from torch.utils.data import DataLoader
+
+from nemo.collections.common.losses import AggregatorLoss, CrossEntropyLoss
+from nemo.collections.common.tokenizers.tokenizer_utils import get_tokenizer
+from nemo.collections.nlp.data.intent_slot_classification import IntentSlotClassificationDataset, IntentSlotDataDesc
+from nemo.collections.nlp.metrics.classification_report import ClassificationReport
+from nemo.collections.nlp.modules.common import SequenceTokenClassifier
+from nemo.collections.nlp.modules.common.common_utils import get_pretrained_lm_model
+from nemo.core.classes import typecheck
+from nemo.core.classes.modelPT import ModelPT
+from nemo.core.neural_types import NeuralType
+from nemo.utils.decorators import experimental
+
+
+@experimental
+class IntentSlotClassificationModel(ModelPT):
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.bert_model.input_types
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.classifier.output_types
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        """ Initializes BERT Joint Intent and Slot model.
+        """
+
+        # init superclass
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        data_desc = IntentSlotDataDesc(data_dir=cfg.data_dir, modes=["train", "test", "dev"])
+
+        # initialize Tokenizer
+        self.tokenizer = get_tokenizer(
+            tokenizer_name=cfg.language_model.tokenizer,
+            pretrained_model_name=cfg.language_model.pretrained_model_name,
+            vocab_file=cfg.language_model.vocab_file,
+            tokenizer_model=cfg.language_model.tokenizer_model,
+            do_lower_case=cfg.language_model.do_lower_case,
+        )
+
+        # initialize Bert model
+        self.bert_model = get_pretrained_lm_model(
+            pretrained_model_name=cfg.language_model.pretrained_model_name,
+            config_file=cfg.language_model.bert_config,
+            checkpoint_file=cfg.language_model.bert_checkpoint,
+        )
+
+        self.data_dir = cfg.data_dir
+        self.max_seq_length = cfg.language_model.max_seq_length
+        self.hidden_size = self.bert_model.config.hidden_size
+
+        self.classifier = SequenceTokenClassifier(
```

Comment:
This code is not going to work. You should have two classification losses here.

Reply:
2 heads and losses are inside of Classifier. Seems working fine.